Using the PackageValidation OOB package results in FileNotFoundExceptions

```
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.packagevalidation\1.0.0-rc.1.21402.3\build\Microsoft.NET.Compatibility.Common.targets(23,5): error MSB4018: The "Microsoft.DotNet.PackageValidation.ValidatePackage" task failed unexpectedly. [C:\git\runtime3\src\libraries\System.Net.Http.Json\src\System.Net.Http.Json.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.packagevalidation\1.0.0-rc.1.21402.3\build\Microsoft.NET.Compatibility.Common.targets(23,5): error MSB4018: System.IO.FileNotFoundException: Could not load file or assembly 'NuGet.Packaging, Version=6.0.0.165, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified. [C:\git\runtime3\src\libraries\System.Net.Http.Json\src\System.Net.Http.Json.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.packagevalidation\1.0.0-rc.1.21402.3\build\Microsoft.NET.Compatibility.Common.targets(23,5): error MSB4018: File name: 'NuGet.Packaging, Version=6.0.0.165, Culture=neutral, PublicKeyToken=31bf3856ad364e35' [C:\git\runtime3\src\libraries\System.Net.Http.Json\src\System.Net.Http.Json.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.packagevalidation\1.0.0-rc.1.21402.3\build\Microsoft.NET.Compatibility.Common.targets(23,5): error MSB4018:    at Microsoft.DotNet.PackageValidation.ValidatePackage.ExecuteCore() [C:\git\runtime3\src\libraries\System.Net.Http.Json\src\System.Net.Http.Json.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.packagevalidation\1.0.0-rc.1.21402.3\build\Microsoft.NET.Compatibility.Common.targets(23,5): error MSB4018:    at Microsoft.NET.Build.Tasks.TaskBase.Execute() in /_/src/Tasks/Common/TaskBase.cs:line 38 [C:\git\runtime3\src\libraries\System.Net.Http.Json\src\System.Net.Http.Json.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.packagevalidation\1.0.0-rc.1.21402.3\build\Microsoft.NET.Compatibility.Common.targets(23,5): error MSB4018:    at Microsoft.DotNet.PackageValidation.ValidatePackage.Execute() in /_/src/Compatibility/Microsoft.DotNet.PackageValidation/ValidatePackage.cs:line 44 [C:\git\runtime3\src\libraries\System.Net.Http.Json\src\System.Net.Http.Json.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.packagevalidation\1.0.0-rc.1.21402.3\build\Microsoft.NET.Compatibility.Common.targets(23,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() in Microsoft.Build.dll:token 0x60015fc+0x3e [C:\git\runtime3\src\libraries\System.Net.Http.Json\src\System.Net.Http.Json.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.packagevalidation\1.0.0-rc.1.21402.3\build\Microsoft.NET.Compatibility.Common.targets(23,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) in Microsoft.Build.dll:token 0x600147b+0x804 [C:\git\runtime3\src\libraries\System.Net.Http.Json\src\System.Net.Http.Json.csproj]
```

I assume this is because the package doesn't carry along the NuGet and MSBuild dependencies.

cc @Anipik @safern @ericstj

--- a/src/Compatibility/Microsoft.DotNet.ApiCompatibility/Microsoft.DotNet.ApiCompatibility.csproj
+++ b/src/Compatibility/Microsoft.DotNet.ApiCompatibility/Microsoft.DotNet.ApiCompatibility.csproj
@@ -9,7 +9,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.CodeAnalysis" Version="$(MicrosoftCodeAnalysisCSharpPackageVersion)" />
+    <PackageReference Include="Microsoft.CodeAnalysis" Version="4.0.0-1.21304.5" />
   </ItemGroup>
 
   <ItemGroup>

--- a/src/Compatibility/Microsoft.DotNet.ApiCompatibility/Microsoft.DotNet.ApiCompatibility.csproj
+++ b/src/Compatibility/Microsoft.DotNet.ApiCompatibility/Microsoft.DotNet.ApiCompatibility.csproj
@@ -4,11 +4,13 @@
     <TargetFramework>netstandard2.0</TargetFramework>
     <StrongNameKeyId>Open</StrongNameKeyId>
     <VersionPrefix>1.0.0</VersionPrefix>
-    <!--We need to compare ISymbols in a special way (by name) and roslyn symbol comparers take more heuristics into consideration.-->
+    <!-- We need to compare ISymbols in a special way (by name) and roslyn symbol comparers take more heuristics into consideration.-->
     <NoWarn>RS1024;$(NoWarn)</NoWarn>
   </PropertyGroup>
 
   <ItemGroup>
+    <!-- We pin the versions of the assets that ship with the SDK in order to be able to run on older SDKs. 
+    Currently we support .NET 6 Preview5 or later. -->
     <PackageReference Include="Microsoft.CodeAnalysis" Version="4.0.0-1.21304.5" />
   </ItemGroup>
 

--- a/src/Compatibility/Microsoft.DotNet.PackageValidation/Microsoft.DotNet.PackageValidation.csproj
+++ b/src/Compatibility/Microsoft.DotNet.PackageValidation/Microsoft.DotNet.PackageValidation.csproj
@@ -13,6 +13,8 @@
 
   <ItemGroup>
     <Compile Include="..\..\Tasks\Common\**\*.cs" LinkBase="Common" />
+    <!-- We pin the versions of the assets that ship with the SDK in order to be able to run on older SDKs. 
+    Currently we support .NET 6 Preview5 or later. -->
     <PackageReference Include="Microsoft.Build.Framework" Version="15.4.8" ExcludeAssets="Runtime" />
     <PackageReference Include="Microsoft.Build.Utilities.Core" Version="15.4.8" ExcludeAssets="Runtime" />
     <PackageReference Include="NuGet.Frameworks" Version="6.0.0-preview.1.66" />

--- a/src/Compatibility/Microsoft.DotNet.PackageValidation/Microsoft.DotNet.PackageValidation.csproj
+++ b/src/Compatibility/Microsoft.DotNet.PackageValidation/Microsoft.DotNet.PackageValidation.csproj
@@ -13,13 +13,12 @@
 
   <ItemGroup>
     <Compile Include="..\..\Tasks\Common\**\*.cs" LinkBase="Common" />
-    <!-- We pin the versions of the assets that ship with the SDK in order to be able to run on older SDKs. 
-    Currently we support .NET 6 Preview5 or later. -->
-    <PackageReference Include="Microsoft.Build.Framework" Version="15.4.8" ExcludeAssets="Runtime" />
-    <PackageReference Include="Microsoft.Build.Utilities.Core" Version="15.4.8" ExcludeAssets="Runtime" />
-    <PackageReference Include="NuGet.Frameworks" Version="6.0.0-preview.1.66" />
-    <PackageReference Include="NuGet.Packaging" Version="6.0.0-preview.1.66" />
-    <PackageReference Include="NuGet.Protocol" Version="6.0.0-preview.1.66" />
+    <PackageReference Include="Microsoft.Build.Framework" Version="$(MicrosoftBuildFrameworkVersion)" ExcludeAssets="Runtime" />
+    <PackageReference Include="Microsoft.Build.Utilities.Core" Version="$(MicrosoftBuildUtilitiesCoreVersion)" ExcludeAssets="Runtime" />
+    <!-- We carry NuGet as part of the package in case the package is used with an older SDKs or with full framework MSBuild. -->
+    <PackageReference Include="NuGet.Frameworks" Version="$(NuGetBuildTasksPackageVersion)" />
+    <PackageReference Include="NuGet.Packaging" Version="$(NuGetBuildTasksPackageVersion)" />
+    <PackageReference Include="NuGet.Protocol" Version="$(NuGetBuildTasksPackageVersion)" />
     <PackageReference Include="System.IO.Compression" Version="4.3.0" />
     <ProjectReference Include="..\Microsoft.DotNet.ApiCompatibility\Microsoft.DotNet.ApiCompatibility.csproj" ExcludeAssets="Runtime" />
   </ItemGroup>

--- a/src/Compatibility/Microsoft.DotNet.PackageValidation/Microsoft.DotNet.PackageValidation.csproj
+++ b/src/Compatibility/Microsoft.DotNet.PackageValidation/Microsoft.DotNet.PackageValidation.csproj
@@ -19,7 +19,7 @@
     <PackageReference Include="NuGet.Frameworks" Version="$(NuGetBuildTasksPackageVersion)" />
     <PackageReference Include="NuGet.Packaging" Version="$(NuGetBuildTasksPackageVersion)" />
     <PackageReference Include="NuGet.Protocol" Version="$(NuGetBuildTasksPackageVersion)" />
-    <ProjectReference Include="..\Microsoft.DotNet.ApiCompatibility\Microsoft.DotNet.ApiCompatibility.csproj" ExcludeAssets="Runtime" />
+    <ProjectReference Include="..\Microsoft.DotNet.ApiCompatibility\Microsoft.DotNet.ApiCompatibility.csproj" />
   </ItemGroup>
 
   <ItemGroup>

--- a/src/Compatibility/Microsoft.DotNet.PackageValidation/Microsoft.DotNet.PackageValidation.csproj
+++ b/src/Compatibility/Microsoft.DotNet.PackageValidation/Microsoft.DotNet.PackageValidation.csproj
@@ -19,7 +19,6 @@
     <PackageReference Include="NuGet.Frameworks" Version="$(NuGetBuildTasksPackageVersion)" />
     <PackageReference Include="NuGet.Packaging" Version="$(NuGetBuildTasksPackageVersion)" />
     <PackageReference Include="NuGet.Protocol" Version="$(NuGetBuildTasksPackageVersion)" />
-    <PackageReference Include="System.IO.Compression" Version="4.3.0" />
     <ProjectReference Include="..\Microsoft.DotNet.ApiCompatibility\Microsoft.DotNet.ApiCompatibility.csproj" ExcludeAssets="Runtime" />
   </ItemGroup>
 

--- a/src/Compatibility/Microsoft.DotNet.PackageValidation/Microsoft.DotNet.PackageValidation.csproj
+++ b/src/Compatibility/Microsoft.DotNet.PackageValidation/Microsoft.DotNet.PackageValidation.csproj
@@ -13,11 +13,11 @@
 
   <ItemGroup>
     <Compile Include="..\..\Tasks\Common\**\*.cs" LinkBase="Common" />
-    <PackageReference Include="Microsoft.Build.Framework" Version="$(MicrosoftBuildFrameworkVersion)" ExcludeAssets="Runtime" />
-    <PackageReference Include="Microsoft.Build.Utilities.Core" Version="$(MicrosoftBuildUtilitiesCoreVersion)" ExcludeAssets="Runtime" />
-    <PackageReference Include="NuGet.Frameworks" Version="$(NuGetBuildTasksPackageVersion)" ExcludeAssets="Runtime" />
-    <PackageReference Include="NuGet.Packaging" Version="$(NuGetBuildTasksPackageVersion)" ExcludeAssets="Runtime" />
-    <PackageReference Include="NuGet.Protocol" Version="$(NuGetBuildTasksPackageVersion)" ExcludeAssets="Runtime" />
+    <PackageReference Include="Microsoft.Build.Framework" Version="15.4.8" ExcludeAssets="Runtime" />
+    <PackageReference Include="Microsoft.Build.Utilities.Core" Version="15.4.8" ExcludeAssets="Runtime" />
+    <PackageReference Include="NuGet.Frameworks" Version="6.0.0-preview.1.66" />
+    <PackageReference Include="NuGet.Packaging" Version="6.0.0-preview.1.66" />
+    <PackageReference Include="NuGet.Protocol" Version="6.0.0-preview.1.66" />
     <PackageReference Include="System.IO.Compression" Version="4.3.0" />
     <ProjectReference Include="..\Microsoft.DotNet.ApiCompatibility\Microsoft.DotNet.ApiCompatibility.csproj" ExcludeAssets="Runtime" />
   </ItemGroup>

